contracts/user-contracts/src/main/java/org/java/mentorship/contracts/user/client/UserFeignClient.java

```diff
@@ -16,4 +20,48 @@ public interface UserFeignClient {
             value = "/users"
     )
     List<User> getUsers();
+
+    @RequestMapping(
+            method = RequestMethod.GET,
+            value = "/users/{id}"
+    )
+    User getUser(@PathVariable("id") Integer id);
+
+    @RequestMapping(
+            method = RequestMethod.POST,
+            value = "/register"
+    )
+    User registerUser(@RequestBody UserRegistrationRequest registrationRequest);
+
+    @RequestMapping(
+            method = RequestMethod.POST,
+            value = "/verify/{id}/{token}"
+    )
+    Boolean verifyUser(@PathVariable("id") Integer id, @PathVariable("token") String token);
+
+
+    @RequestMapping(
+            method = RequestMethod.POST,
+            value = "/sessions"
+    )
+    Session createSession(@RequestBody UserLoginRequest loginRequest);
+
+    @RequestMapping(
+            method = RequestMethod.GET,
+            value = "/sessions/{key}"
+    )
+    Session getSession(@PathVariable("key") String key);
+
+    @RequestMapping(
+            method = RequestMethod.GET,
+            value = "/sessions/user/{id}"
```

Comment:
Not quite restful

To get sessions by userId, you'd do GET @ /sessions?userId={id}

Reply:
This still holds true
Any chance you could modify the path?